### PR TITLE
PRs with docs changes appear in their own sections in changelog

### PR DESCRIPTION
**Pull request by haywoodsh**
### Proposed changes

This PR rearranges the labels for the auto-generated change log so that PRs with docs changes appear in their own sections instead of being grouped under “📝 Documentation”. The current order of labels causes PRs with other labels (such as helm chart, chore, tests) to be placed under “📝 Documentation” if they also have documentation changes. This makes the change log less accurate and informative about the work done in a release. For example, see https://github.com/nginxinc/kubernetes-ingress/releases/tag/v3.0.0. 

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [ ] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by ciarams87 — COMMENTED** [review]
Looks good!

**Review by ciarams87 — APPROVED** [review]

**Review by jjngx — APPROVED** [review]

**Review by vepatel — APPROVED** [review]

File: .github/release.yml [hunk 1]
```diff
@@ -27,6 +24,9 @@ changelog:
     - title: ⬆️ Dependencies
       labels:
         - dependencies
+    - title: 📝 Documentation
```
Comment (lucacome) [inline .github/release.yml:27]: I think we probably want to have docs before dependencies. Usually the list of dependency updates is very long, making it hard to see the docs at the end of it.

Comment (lucacome) [reply .github/release.yml:27]: I was thinking that we should probably also re-order `change`, and `enhancement` in order of importance, something like:
```
    - title: 💣 Breaking Changes
      labels:
        - change
    - title: 🚀 Features
      labels:
        - enhancement
    - title: 🐛 Bug Fixes
      labels:
        - bug
    - title: 📦 Helm Chart
      labels:
        - helm_chart
    - title: 🧪 Tests
      labels:
        - tests
    - title: 🔨 Maintenance
      labels:
        - chore
    - title: 📝 Documentation
      labels:
        - documentation
    - title: ⬆️ Dependencies
      labels:
        - dependencies
    - title: Other Changes
      labels:
        - "*"
 ```
what do you think?

Comment (jasonwilliams14) [reply .github/release.yml:27]: I think this looks good. This puts 👀 directly on items that users should be aware of that might get looked over it is lower down the list.
brings the attention to any breaking changes and new features we implement.

**Review by lucacome — APPROVED** [review]

**Pull request merged**